### PR TITLE
Introduce an init_app(app) for factory pattern

### LINKED ISSUE #24: Document how to use Flask decorator with AppFactory/Blueprint pattern
I had to do this in my create_app function-

    with app.app_context():
        app.auth = Auth(
            app,
            authority=os.getenv("AUTHORITY"),
            client_id=os.getenv("CLIENT_ID"),
            client_credential=os.getenv("CLIENT_SECRET"),
            redirect_uri=os.getenv("REDIRECT_URI"),
        )

And then make a wrapper decorator in my Blueprint:
  
    def login_required(f):
        """Decorator to require login for a route."""
        @wraps(f)
        def decorated_function(*args, **kwargs):
            return current_app.auth.login_required(f)(*args, **kwargs)
        return decorated_function

### PR DESCRIPTION
**Pull request by rayluo**
This proof-of-concept may close #24. See if that meets your need, @pamelafox . If it looks good, I will see if I can apply same changes to Quart.

### CODE REVIEW

File: identity/flask.py [hunk 1]
```diff
@@ -18,19 +18,56 @@
...
+                auth = Auth(app, authority=..., client_id=..., ...)
+
+            2. But if you are using `Application Factory pattern
+            <https://flask.palletsprojects.com/en/latest/patterns/appfactories/>`_,
+            your app is not available globally, so you need to pass ``None`` here,
+            and call :func:`Auth.init_app()` later,
+            inside or after your app factory function. Usage::
+
+                # In your auth.py
+                auth = Auth(None, authority=..., client_id=..., ...)
```
Comment (pamelafox) [inline identity/flask.py]:
```suggestion
                auth = Auth(app=None, authority=..., client_id=..., ...)
```

I think that looks nicer when passing in None

**Review by pamelafox — APPROVED** [review]
Nice! The init_app method is how I often initiate other extensions, so this seems right? I assume you've tried this out in a Flask app.

**Comment by joaooliveira-proof** [comment]
I've tested in my app, worked like a charm. Except that the linter did not like that I was passing None to Auth.

**Comment by rayluo** [comment]
Polished the previous proof-of-concept (PoC), and applied it to Quart, manually tested with my two samples (which will be pushed soon in their own PRs in different repos).

@joaooliveira-proof , if you are still interested, please pull the latest branch and test it again. It shall be a drop-in replacement of the previous PoC, except that the linter shall be happy now. :-)

**Pull request merged**